src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -116,6 +121,10 @@ private String setPlayerTeamToTeleporterTeam(EntityRef player, EntityRef telepor
     private void handlePlayerTeleport(EntityRef player, String team) {
         Vector3f randomVector = new Vector3f(-1 + random.nextInt(3), 0, -1 + random.nextInt(3));
         player.send(new CharacterTeleportEvent(randomVector.add(LASUtils.getTeleportDestination(team))));
+        Optional<Prefab> prefab = Assets.getPrefab("inventory");
+        StartingInventoryComponent startingInventory = prefab.get().getComponent(StartingInventoryComponent.class);
+        player.addComponent(startingInventory);
+        player.send(new RequestInventoryEvent());
         inventoryManager.giveItem(player, EntityRef.NULL, entityManager.create(LASUtils.MAGIC_STAFF_URI));
```

Comment:
Why is the magic staff added separately and not part of the respawn inventory?

Reply:
The magic staff isn't dropped on death so the player get's it back on respawn. Over here we are using the same inventory for spawn and respawn and that would mean that on respawn the player will have multiple staffs, I could create separate inventories for spawn and respawn but most of the weapons are similar currently and changes to the inventory will be coming later on.